### PR TITLE
Add /pa endpoints (CU-21ghk0x) and upgrade NodeJS from 12 to 16

### PR DESCRIPTION
**Pull request by trisapeace**
- We were using Node 12, which just reached end-of-life. So upgrading Travis and our Docker container to use 16.14.2, which is the version in active long term support (https://nodejs.org/en/about/releases/). This is a big jump and basically means that we'll need to test everything.

## Test Plan
- :heavy_check_mark: Deploy on Sensors Dev
- :heavy_check_mark: Run smoke tests
- :heavy_check_mark: /pa endpoint tests:
  - :heavy_check_mark: Phone Number tests
    - :heavy_check_mark: Use PA to buy a Sensors phone number
       - :heavy_check_mark: Is in the Sensors-Production Twilio account
       - :heavy_check_mark: Has given area code
       - :heavy_check_mark: Has given friendly name
       - :heavy_check_mark: Has this the Sensors Dev webhook URL
       - :heavy_check_mark: Has the All Sensor Messages messaging service
    - :heavy_check_mark: Provide an invalid area code (logs the error in the backend. It would be nice if this showed the error somewhere in the front end, but this is OK for now)
    - :heavy_check_mark: Provide a valid area code that has no available numbers (logs the error in the backend. It would be nice if this showed the error somewhere in the front end, but this is OK for now)
    - :heavy_check_mark: Provide a duplicate friendly name (this succeeds)
    - :heavy_check_mark: Use Postman to provide a missing, invalid or blank ClickUp token (returns 401)
    - :heavy_check_mark: Use Postman to provide a missing, invalid or blank API key (returns 401)
    - :heavy_check_mark: Use PA to provide a ClickUp token from a different account (with and without a project called "Brave Technology Cooperative") (returns 401)
  - :heavy_check_mark: Use Postman to test API key rotation
     - :heavy_check_mark: Primary API key works (checked Get Clients endpoint)
     - :heavy_check_mark: Secondary API key works (checked Get Clients endpoint)
     - :heavy_check_mark: Other values return 401 Unauthorized (checked Get Clients endpoint)
  - :heavy_check_mark: Password rotation
     - :heavy_check_mark: Primary password works (checked Create Location endpoint)
     - :heavy_check_mark: Secondary password works (checked Create Location endpoint)
     - :heavy_check_mark: Other values return 401 Unauthorized (checked Create Location endpoint)
  - :heavy_check_mark: Use PA to Create Sensors location in the DB
     - :heavy_check_mark: location created in DB with valid values from the form (right now, this doesn't include a Twilio Number. We should add that in the future)
     - :heavy_check_mark: What if I put in invalid values? Error handling? (not a whole lot, if the actual DB insert fails, then it will return 401)
    - :heavy_check_mark: Use Postman to provide a missing, invalid or blank ClickUp token
    - :heavy_check_mark: Use Postman to provide a missing, invalid or blank API key
    - :heavy_check_mark: Use Postman to provide a missing, invalid or blank password
  - :heavy_check_mark: Use PA to retrieve Sensors client list from the DB
     - :heavy_check_mark: All clients (active and inacitve) are listed, in alphabetical order
    - :heavy_check_mark: Use Postman to provide a missing, invalid or blank ClickUp token (returns 401)
    - :heavy_check_mark: Use Postman to provide a missing, invalid or blank API key (returns 401)
- :heavy_check_mark: Run through as many other tests as I can think of
   - :heavy_check_mark: Real XeThru+SSM  
      - :heavy_check_mark: Stillness alert
      - :heavy_check_mark: Duration alert
      - :heavy_check_mark: Respond to all the messages
      - :heavy_check_mark: See reminder and fallback messages
      - :heavy_check_mark: Disconnection/reconnection messages
   - :heavy_check_mark: Real INS+FSM
      - :heavy_check_mark: Stillness alert
      - :heavy_check_mark: Duration alert
      - :heavy_check_mark: Respond to all the messages
      - :heavy_check_mark: See reminder and fallback messages
      - :heavy_check_mark: Disconnection/reconnection messages
  - :heavy_check_mark: Dashboard
     - :heavy_check_mark: See Stillness and Duration alerts
     - :heavy_check_mark: CreatedAt, RespondedAt, and UpdatedAt update as expected
     - :heavy_check_mark: Incident category displayed as chosen
     - :heavy_check_mark: Can view vitals
     - :heavy_check_mark: Update client (to change active/inactive status)
     - :heavy_check_mark: Update location (to change duration and stillness thresholds)
     - :heavy_check_mark: New client
     - :heavy_check_mark: New location
     - :heavy_check_mark: Export CSV
     - :heavy_check_mark: Login (valid and invalid username/password)
     - :heavy_check_mark: Logout
     - :heavy_check_mark: Can't access internal pages while logged out

### CODE REVIEW

**Comment by trisapeace** [comment]
Task linked: [CU-21ghk0x Further streamline the sensor provisioning process with the (Particle Accelerator)](https://app.clickup.com/t/21ghk0x)

**Comment by trisapeace** [comment]
@reesecritchlow This one also isn't entirely done yet because I haven't tested it (or even finished writing out my test cases). But if you want to take a look at it now, you can see how I've refactored your endpoints into the `routes.js` and `pa.js` file as well as some moved into `brave-alert-lib` because they are useful for both the Buttons and Sensors code that you wrote.

File: server/.env.example [hunk 1]
```diff
@@ -93,3 +98,29 @@ LOW_BATTERY_ALERT_TIMEOUT_TEST = 2
 # Login to our OneSignal Account. Get this value from Settings --> Messaging --> Android Categories --> Alerts and Reminders --> Channel ID
 ONESIGNAL_ALERT_ANDROID_CATEGORY_ID=guid-here
 ONESIGNAL_ALERT_ANDROID_CATEGORY_ID_TEST=android-group-guid
+
+# Name of the ClickUp Team where the Sensor Tracker lives
+# Get this from Clickup --> Settings --> Workspace Settings
```
Comment (reesecritchlow) [inline server/.env.example:103]: Might also want to note that this is also Brave's Team/Workspace, as that is the more the principle that this is used for in terms of security.

File: server/.env.example [hunk 2]
```diff
@@ -93,3 +98,29 @@ LOW_BATTERY_ALERT_TIMEOUT_TEST = 2
...
+PA_API_KEY_SECONDARY=MyLongKeyBackup
+PA_API_KEY_SECONDARY_TEST=MyLongKeyBackup
+
+# Primary and Secondary passwords used by PA before making any calls that will affect the DB
+# Two passwords are used to allow for seamless password rotation
+# Get this from 1Password --> Brave Sensor Credentials --> PA Dashboard Renamer password
+PA_PASSWORD_PRIMARY=Apassword
+PA_PASSWORD_PRIMARY_TEST=Apassword
+PA_PASSWORD_SECONDARY=ABpassword
+PA_PASSWORD_SECONDARY_TEST=ABpassword
```
Comment (reesecritchlow) [inline server/.env.example:126]: I'm unsure about the general style of a .env.example file, but should there be any instructions here on what to change in the frontend code when these keys/passwords get rotated?

File: server/routes.js [hunk 3]
```diff
@@ -27,6 +30,10 @@ function configureRoutes(app) {
   app.post('/api/sirenAddressed', siren.validateSirenAddressed, siren.handleSirenAddressed)
   app.post('/api/sirenEscalated', siren.validateSirenEscalated, siren.handleSirenEscalated)
 
+  app.post('/pa/create-sensor-location', clickUpHelpers.clickUpChecker, pa.handleCreateSensorLocation)
```
Comment (reesecritchlow) [inline server/routes.js:33]: I haven't seen this way of using express before, and can't seem to quite put it into words to find documentation on it. Out of curiosity, does it just call the functions successively until a response is sent? I like how it's infinitely cleaner.

**Review by reesecritchlow — COMMENTED** [review]
Again, looks good, really appreciate the help and also the modularity/cleanliness of the code in this migration.

File: server/.env.example (hunk 1 above)
Comment (trisapeace) [reply server/.env.example:103]: I was just avoiding putting the full name. But I can make it a little clearer.

File: server/routes.js (hunk 3 above)
Comment (trisapeace) [reply server/routes.js:33]: Here's some good documentation: https://expressjs.com/en/guide/using-middleware.html

File: server/.env.example (hunk 1 above)
Comment (trisapeace) [reply server/.env.example]: Updated!

File: server/package.json [hunk 4]
```diff
@@ -18,7 +18,7 @@
     "@sentry/node": "^6.2.5",
     "@sentry/tracing": "^6.2.5",
     "axios": "^0.21.2",
-    "brave-alert-lib": "https://github.com/bravetechnologycoop/brave-alert-lib#v6.6.0",
+    "brave-alert-lib": "https://github.com/bravetechnologycoop/brave-alert-lib#td-add-pa-endpoints",
```
Comment (trisapeace) [inline server/package.json:21]: Reminder to update this when it has a real version number

Comment (trisapeace) [reply server/package.json]: Done.

**Pull request merged**